path/windows/to_file_url.ts

```diff
@@ -20,10 +20,12 @@ export function toFileUrl(path: string): URL {
   if (!isAbsolute(path)) {
     throw new TypeError("Must be an absolute path.");
   }
-
   const [, hostname, pathname] = path.match(
     /^(?:[/\\]{2}([^/\\]+)(?=[/\\](?:[^/\\]|$)))?(.*)/,
   )!;
+  if (!pathname) {
+    throw new TypeError("Invalid pathname.");
+  }
   const url = new URL("file:///");
   url.pathname = encodeWhitespace(pathname.replace(/%/g, "%25"));
```

Comment:
to use `pathname.replace()` first it's needed to check that pathname is not `undefined` (due to maybe a failed regex match)

Reply:
Are you able to add a test for this new case?